api-module-library/hubspot/tests/api.test.js

```diff
@@ -1,26 +1,376 @@
+            expect(results[1].properties.name).toBe('Gitlab');
+            expect(results[1].properties.domain).toBe('gitlab.com');
+        });
+
+        it('should update a Batch of Companies', async () => {
+            let body = [
+                {
+                    properties: {
+                        name: 'Facebook 2',
+                    },
+                    id: createResponse.results[0].id,
+                },
+                {
+                    properties: {
+                        name: 'Gitlab 2',
+                    },
+                    id: createResponse.results[1].id,
+                },
+            ];
+            let response = await api.updateBatchCompany(body);
+
+            let results = _.sortBy(response.results, [
+                function (o) {
+                    return o.properties.name;
+                },
+            ]);
+            expect(response.status).toBe('COMPLETE');
+            expect(results[0].properties.name).toBe('Facebook 2');
+            expect(results[1].properties.name).toBe('Gitlab 2');
+        });
     });
 
-    it('should return auth requirements', async () => {
-        const requirements = await manager.getAuthorizationRequirements();
-        expect(requirements).exists;
-        expect(requirements.type).toEqual('oauth2');
+    describe('HS Contacts', () => {
+        let createResponse;
+        beforeAll(async () => {
+            let body = {
+                email: 'jose.miguel@hubspot.com',
+                firstname: 'Miguel',
+                lastname: 'Delgado',
+            };
+            createResponse = await api.createContact(body);
+        });
+
+        it('should create a Contact', async () => {
+            expect(createResponse).toHaveProperty('id');
+            expect(createResponse.properties.firstname).toBe('Miguel');
+            expect(createResponse.properties.lastname).toBe('Delgado');
+        });
+
+        it('should list Contacts', async () => {
+            let response = await api.listContacts();
+            expect(response.results[0]).toHaveProperty('id');
+            expect(response.results[0]).toHaveProperty('properties');
+            expect(response.results[0].properties).toHaveProperty('firstname');
+        });
+
+        it.skip('should update a Contact', async () => {
+            let body = {
+                lastname: 'Johnson (Sample Contact) 1',
+            };
+            let response = await api.updateContact(
+                body,
+                createResponse.id
+            );
+            expect(response.properties.lastname).toBe(
+                'Johnson (Sample Contact) 1'
+            );
+        });
+
+        it('should delete a contact', async () => {
+            let response = await api.archiveContact(createResponse.id);
+            expect(response.status).toBe(204);
+        });
+    });
+
+    describe.skip('HS Contacts BATCH', () => {
```

Comment:
Info: why are we skipping this entire block?

Reply:
added comment to indicate the need for refactor